tests/test_client_helpers.py

```diff
@@ -4,37 +4,84 @@
 from typing import Any
 
 import pytest
+from lakefs_sdk.exceptions import ApiException
 
 import lakefs_spec.client_helpers as client_helpers
 from lakefs_spec import LakeFSFileSystem
-from tests.util import RandomFileFactory
+from tests.util import RandomFileFactory, commit_random_file_on_branch
 
 
 def test_create_tag(
     random_file_factory: RandomFileFactory, fs: LakeFSFileSystem, repository: str, temp_branch: str
 ) -> None:
-    random_file = random_file_factory.make()
-    lpath = str(random_file)
-    rpath = f"{repository}/{temp_branch}/{random_file.name}"
+    commit_random_file_on_branch(
+        random_file_factory=random_file_factory,
+        fs=fs,
+        repository=repository,
+        temp_branch=temp_branch,
+    )
 
-    fs.put(lpath, rpath, precheck=False)
+    tagname = f"Change_{uuid.uuid4()}"
+    try:
+        tag = client_helpers.create_tag(
+            client=fs.client, repository=repository, ref=temp_branch, tag=tagname
+        )
 
-    client_helpers.commit(
-        client=fs.client, repository=repository, branch=temp_branch, message="Commit File Factory"
+        assert tag in client_helpers.list_tags(fs.client, repository)
+        existing_tag = client_helpers.create_tag(
+            client=fs.client, repository=repository, ref=temp_branch, tag=tagname
+        )
+        assert tag == existing_tag
+    finally:
+        fs.client.tags_api.delete_tag(repository=repository, tag=tagname)
+
+
+def test_cannot_reassign_tag(
+    random_file_factory: RandomFileFactory, fs: LakeFSFileSystem, repository: str, temp_branch: str
+) -> None:
+    commit_random_file_on_branch(
+        random_file_factory=random_file_factory,
+        fs=fs,
+        repository=repository,
+        temp_branch=temp_branch,
     )
 
-    tag = f"Change_{uuid.uuid4()}"
+    tagname = f"Change_{uuid.uuid4()}"
     try:
-        new_tag = client_helpers.create_tag(
-            client=fs.client, repository=repository, ref=temp_branch, tag=tag
+        tag = client_helpers.create_tag(
+            client=fs.client, repository=repository, ref=temp_branch, tag=tagname
         )
-        assert tag in [commit.id for commit in client_helpers.list_tags(fs.client, repository)]
-        existing_tag = client_helpers.create_tag(
-            client=fs.client, repository=repository, ref=temp_branch, tag=tag
+        commit_random_file_on_branch(
+            random_file_factory=random_file_factory,
+            fs=fs,
+            repository=repository,
+            temp_branch=temp_branch,
         )
-        assert new_tag == existing_tag
+        with pytest.raises(ApiException):
```

Comment:
I assume that the error here will be a "tag already exists" error, have you verified this? What error message does this produce?

Reply:
Btw, 
the terminal outputs the following:
```
raise ApiException(http_resp=r)
E           lakefs_sdk.exceptions.ApiException: (409)
E           Reason: Conflict
E           HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'X-Request-Id': '19a26127-d216-4e93-af9b-45aff9e6a940', 'Date': 'Tue, 28 Nov 2023 19:08:34 GMT', 'Content-Length': '45'})
E           HTTP response body: {"message":"tag already exists: not unique"}

.venv/lib/python3.11/site-packages/lakefs_sdk/rest.py:237: ApiException
```

Reply:
Fair. Can you `assert e.status == 409`?